packages/core/src/internal/runSaga.js

```diff
@@ -73,6 +76,11 @@ export function runSaga(options, saga, ...args) {
     }
   }
 
+  const effectRunnerMap = {}
+  object.assign(effectRunnerMap, customEffectRunnerMap)
+  // core-effect-runners have higher priority over custom-effect-runners
+  object.assign(effectRunnerMap, coreEffectRunnerMap)
```

Comment:
it still can be changed in user land, since `effectRunnerMap` is available via `env`. should we freeze object?

Reply:
I think it is enough that we document `env` should never be mutated.  Custom effects are advanced usages of redux-saga (if we plan to support custom effects), and users must take care of their effect runners.